sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/VpnSite.java

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.network.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.management.SubResource;
+import com.azure.resourcemanager.network.NetworkManager;
+import com.azure.resourcemanager.network.fluent.models.VpnSiteInner;
+import com.azure.resourcemanager.network.fluent.models.VpnSiteLinkInner;
+import com.azure.resourcemanager.resources.fluentcore.arm.models.GroupableResource;
+import com.azure.resourcemanager.resources.fluentcore.arm.models.Resource;
+import com.azure.resourcemanager.resources.fluentcore.model.Appliable;
+import com.azure.resourcemanager.resources.fluentcore.model.Creatable;
+import com.azure.resourcemanager.resources.fluentcore.model.Refreshable;
+import com.azure.resourcemanager.resources.fluentcore.model.Updatable;
+
+import java.util.List;
+
+/**
+ * Entry point for Virtual Network management API in Azure.
+ */
+@Fluent
+public interface VpnSite
+    extends GroupableResource<NetworkManager, VpnSiteInner>, Refreshable<VpnSite>, Updatable<VpnSite.Update> {
+
+    /** @return returns true if current VPN site is a security site. */
+    boolean isSecuritySite();
+
+    /** @return the AddressPrefixes value of VPN site. */
+    List<String> addressPrefixes();
+
+    /** @return the virtualWan property of VPN site. */
+    SubResource virtualWan();
```

Comment:
Don't directly return `SubResource`. 
Either we return string(`virtualWanId()`), or fluent interface, e.g. `VirtualWan`.
Does customer need this property?

Reply:
Fixed in the new version.